### PR TITLE
Updates codecov.io GH action

### LINKED ISSUE #851: CodeCov uploads are failing
The repo's code coverage statistics haven't been updated in a while.

It appears that the upload fails on the first CI Action attempt, then reports success (while quietly failing) on the second attempt because it couldn't find the upload from the first commit to compare it to.

Here's [an example of a failure](https://github.com/amazon-ion/ion-rust/actions/runs/11600675519/job/32301664122#step:6:34).

image

The suggested fix is to "upload with the Codecov repository upload token to resolve issue":
> [2024-10-30T19:59:23.227Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 429 - {"message":"Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 962s."}

### PR DESCRIPTION
**Pull request by zslayton**
Fixes #851.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -14,33 +14,21 @@ jobs:
...
-      - name: Git Checkout
-        uses: actions/checkout@v3
+      - name: Git checkout
+        uses: actions/checkout@v4
         with:
           submodules: recursive
-      - name: Rust Toolchain
-        uses: actions-rs/toolchain@v1
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
```
Comment (zslayton) [inline .github/workflows/coverage.yml:22]: 🗺️ The GH Actions owned by the `actions-rs` organization are now [archived and unmaintained](https://github.com/actions-rs). `dtolnay/rust-toolchain` is its successor.

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -14,33 +14,21 @@ jobs:
...
-          args: --verbose --workspace --all-features --no-fail-fast --codecov --output-path codecov.json
-      - name: Codecov Upload
+      - name: Install `cargo-llvm-cov`
+        run: cargo install cargo-llvm-cov
+      - name: Cargo test w/code coverage
+        run: cargo llvm-cov --verbose --workspace --all-features --no-fail-fast --codecov --output-path codecov.json
+      - name: Codecov upload
         uses: codecov/codecov-action@v4
-        env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (zslayton) [inline .github/workflows/coverage.yml:43]: 🗺️ Removing this as #852 added `with: token` on line 35 below (hidden by default in this diff).

https://github.com/amazon-ion/ion-rust/pull/853/files#diff-a2115d277b5ca5a2f09a999e53440839cf332b94da177f3d1766334555b0f7c6L44-L47

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -14,33 +14,21 @@ jobs:
...
+        uses: dtolnay/rust-toolchain@stable
         with:
-          profile: minimal
-          toolchain: stable
           # We need this component to generate source coverage in stable
           components: llvm-tools-preview
-          override: true
-      # Wrapper for running coverage with -C instrument-coverage
-      - name: Cargo Install Coverage Wrapper
-        uses: actions-rs/cargo@v1
```
Comment (zslayton) [inline .github/workflows/coverage.yml:31]: 🗺️ The `actions-rs/cargo` action was a pretty thin layer around simple `cargo [subcommand]` operations. I'm not sure how much value it was really adding.

**Review by zslayton — COMMENTED** [review]
🗺️ PR Tour 🧭

**Review by popematt — APPROVED** [review]

**Pull request merged**